app/api/v1/routes/blockchain.py

```diff
@@ -391,11 +391,70 @@ def getNFTBox(tokenId: str, allowCached=False):
             if len(items) == 1:
                 return items[0]
             else:
-                logging.error(f'ERR:{myself()}: multiple nft box ({e})')
+                logging.error(f'ERR:{myself()}: multiple nft box or tokenId doesn\'t exist')
     except Exception as e:
         logging.error(f'ERR:{myself()}: unable to find nft box ({e})')
         return None
 
+# GET unspent boxes by token id direct from explorer db
+def getUnspentStakeKeyTokenBoxes():
+    try:
+        con = create_engine(EXPLORER)
+        sql = f"""
+            -- /unspent/byTokenId (optimized for stakeTokenId)
+            select
+                o.box_id as box_id,
+                -- additional registers in JSON format
+                -- R4 penalty
+                -- R5 stake key
+                o.additional_registers as additional_registers,
+                a.token_id as token_id,
+                -- index of the token in assets list
+                -- 0 stake token
+                -- 1 ergopad staked
+                a.index as index,
+                 -- amount of token in the box
+                a.value as value
+            from
+                node_outputs o
+                join node_assets a on a.box_id = o.box_id
+                and a.header_id = o.header_id
+            where
+                o.box_id in (
+                    -- sub query to get unspent box ids
+                    -- non correlated sub query => executed once
+                    select
+                        o.box_id as box_id
+                    from
+                        node_outputs o
+                        left join node_inputs i on i.box_id = o.box_id
+                        join node_assets a on a.box_id = o.box_id
+                        and a.header_id = o.header_id
+                    where
+                        o.main_chain = true
+                        and i.box_id is null -- output with no input = unspent
+                        and a.token_id = '1028de73d018f0c9a374b71555c5b8f1390994f2f41633e7b9d68f77735782ee' -- stake key token id
```

Comment:
not sure how the indexes are on these tables but probably another performance improvement could be adding an address constraint. Stake boxes are always at:
3eiC8caSy3jiCxCmdsiFNFJ1Ykppmsmff2TEpSsXY1Ha7xbpB923Uv2midKVVkxL3CzGbSS2QURhbHMzP9b9rQUKapP1wpUQYPpH8UebbqVFHJYrSwM3zaNEkBkM9RjjPxHCeHtTnmoun7wzjajrikVFZiWurGTPqNnd1prXnASYh7fd9E2Limc2Zeux4UxjPsLc1i3F9gSjMeSJGZv3SNxrtV14dgPGB9mY1YdziKaaqDVV2Lgq3BJC9eH8a3kqu7kmDygFomy3DiM2hYkippsoAW6bYXL73JMx1tgr462C4d2PE7t83QmNMPzQrD826NZWM2c1kehWB6Y1twd5F9JzEs4Lmd2qJhjQgGg4yyaEG9irTC79pBeGUj98frZv1Aaj6xDmZvM22RtGX5eDBBu2C8GgJw3pUYr3fQuGZj7HKPXFVuk3pSTQRqkWtJvnpc4rfiPYYNpM5wkx6CPenQ39vsdeEi36mDL8Eww6XvyN4cQxzJFcSymATDbQZ1z8yqYSQeeDKF6qCM7ddPr5g5fUzcApepqFrGNg7MqGAs1euvLGHhRk7UoeEpofFfwp3Km5FABdzAsdFR9

Reply:
nice will try this out.

Reply:
nice will try this out.